File: minRotatedSort.java
// Time Complexity : O(logn)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : yes
public class minRotatedSort {
    public int findMin(int[] nums) {
        int low = 0;
        int high = nums.length-1;
        while(low < high){
            int mid = low + (high-low)/2;
            if(nums[mid] > nums[high]){
                low = mid + 1;
            }
            else {
                high = mid;
            }
        }
        return nums[low];
    }
}

File: min_rotatedSort.py
class Solution:
    import math 
    def findMin(self, nums: List[int]) -> int:
        l = 0
        r = len(nums)-1
        res = nums[0]
        while l <=r:
            if nums[r]>nums[l]:
                res = min(res,nums[l])
                break
            mid = (l+r)//2
            res = min(res,nums[mid])
            if nums[mid] >= nums[l]:
                l=mid+1
            else:
                r=mid-1
        return res


            